crates/turborepo-lib/src/run/mod.rs

```diff
@@ -102,13 +114,36 @@ impl<'a> Run<'a> {
             TurboState::platform_name(),
         );
         let start_at = Local::now();
+        let api_auth = self.base.api_auth()?;
+        let api_client = self.base.api_client()?;
+        let (analytics_sender, analytics_handle) =
+            Self::initialize_analytics(api_auth.clone(), api_client.clone()).unzip();
+
+        let result = self
+            .run_with_analytics(start_at, api_auth, api_client, analytics_sender)
+            .await;
+
+        if let Some(analytics_handle) = analytics_handle {
+            analytics_handle.close_with_timeout().await;
+        }
```

Comment:
I think once #6357 lands we'll revert this and instead setup a [cleanup subscriber like we do for the process manager](https://github.com/vercel/turbo/pull/6357/files#diff-c0b6ca703dfc4f10dcb4ccdc330c318feb29f12affc980179d34b45039416727R58-R64)